engine/standalone/util/src/main/scala/pl/touk/nussknacker/engine/standalone/utils/typed/TypingResultToJsonSchemaConverter.scala

```diff
@@ -0,0 +1,39 @@
+package pl.touk.nussknacker.engine.standalone.utils.typed
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.Json
+import io.circe.syntax._
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypedObjectTypingResult, TypingResult}
+import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
+
+class TypingResultToJsonSchemaConverter(exampleValues: Map[String, Json]) extends LazyLogging {
+
+  val jsonEncoder = BestEffortJsonEncoder(failOnUnkown = true, getClass.getClassLoader)
+
+  def typingResultToJsonSchema(typingResult: TypingResult, paramName: Option[String] = None): Json =
+    jsonEncoder.encode(
+      typingResult match {
+        case typed: TypedObjectTypingResult => typed.fields.map(v => v._1 -> typingResultToJsonSchema(v._2, Some(v._1)))
+        case typed: TypingResult => translateTypeResult(typed, paramName.flatMap(exampleValues.get))
+      }
+    )
+
+  private def translateTypeResult(typingResult: TypingResult, example: Option[Json] = None): Map[String, Json] = {
+    (typingResult match {
+      case TypedClass(klass, _) if klass == classOf[java.lang.String] =>
+        Map("type" -> "string".asJson)
+      case TypedClass(klass, _) if klass == classOf[java.lang.Boolean] =>
+        Map("type" -> "boolean".asJson)
+      case TypedClass(klass, _) if klass == classOf[java.lang.Integer] =>
+        Map("type" -> "integer".asJson)
+      case TypedClass(klass, _) if klass == classOf[java.lang.Number] =>
+        Map("type" -> "number".asJson)
+      case TypedClass(klass, params) if klass == classOf[java.util.List[_]] =>
+        Map("type" -> "array".asJson, "items" -> params.flatMap(p => translateTypeResult(p)).toMap.asJson)
+      case TypedClass(klass, params) if klass == classOf[java.util.Map[_, _]] =>
+        Map("type" -> "object".asJson, "additionalProperties" -> params.flatMap(p => translateTypeResult(p)).toMap.asJson)
+      case _ => Map()
```

Comment:
What about e.g. Union, Dicts, etc? Maybe we should pattern match on SingleTypingResult?

Reply:
As mentioned earlier I have removed this file from PR.